### PR TITLE
Support show knowhere prometheus metrics

### PR DESCRIPTION
**Pull request by cydrain**

### CODE REVIEW

File: internal/metrics/metrics.go [hunk 1]
```diff
@@ -102,3 +102,11 @@ func Register(r *prometheus.Registry) {
 		Handler: promhttp.Handler(),
 	})
 }
+
+// CRegister serves prometheus http service
+func CRegister(r *CRegistry) {
+	management.Register(&management.HTTPHandler{
+		Path:    "/metrics_c",
```
[the remark this answers is not in the document]
Comment (cydrain) [reply internal/metrics/metrics.go:109]: CRegistry is different with prometheus.Registry, it re-write API Gather()

File: cmd/roles/roles.go [hunk 2]
```diff
@@ -49,11 +49,20 @@ import (
...
 	Registry = prometheus.NewRegistry()
 	Registry.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
 	Registry.MustRegister(prometheus.NewGoCollector())
 	metrics.RegisterEtcdMetrics(Registry)
+
+	CRegistry = &metrics.CRegistry{
+		Registry: prometheus.NewRegistry(),
+	}
+	CRegistry.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+	CRegistry.MustRegister(prometheus.NewGoCollector())
```
Comment (LoveEachDay) [inline cmd/roles/roles.go:65]: CRegistry doesn't need go process and go gc collector.

File: internal/metrics/metrics.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (LoveEachDay) [reply internal/metrics/metrics.go:109]: Here I propose a wrapper for Registry and CRegistry like this:
```
type Registry struct {
  GRegistry *prometheus.Registry
  CRegistry *metrics.CRegistry
}
```
Here `GRegistry` will responsible for Go metrics, and `CRegistry` will responsible for C++ related metrics, and we'll implement a `Gather` function like this:
```
func (r *Registry) Gather() (res []*dto.MetricFamily, err error) {
  mf1, err := r.GRegistry.Gather()
  mf2, err := r.CRegistry.Gather()
  mf1 = append(mf1, mf2)
  return mf1, errr
}
```

File: cmd/roles/roles.go (hunk 2 above)
Comment (cydrain) [reply cmd/roles/roles.go]: ok

File: internal/metrics/metrics.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cydrain) [reply internal/metrics/metrics.go]: good idea

**Comment by Bennu-Li** [comment]
rerun ut

**Review by congqixia — COMMENTED** [review]
/approve

**Pull request merged**